bpf/lib/common.h

```diff
@@ -724,6 +724,13 @@ enum metric_dir {
 #define MARK_MAGIC_TO_PROXY		0x0200
 #define MARK_MAGIC_SNAT_DONE		0x0300
 #define MARK_MAGIC_OVERLAY		0x0400
+/* used to indicate encrypted traffic was tunnel encapsulated
+ * this is useful in the IPsec code paths where we need to know if overlay
+ * traffic is encrypted or not.
+ *
+ * the SPI bit can be reused since this magic mark is only used POST encryption.
+ */
+#define MARK_MAGIC_OVERLAY_ENCRYPTED	0x1400
```

Comment:
wow, that's a lot of smart games with aliasing `MARK_MAGIC_OVERLAY` in (hopefully) all the right places :grin:. This can go away again in `v1.19`, correct?

Just a nit - could you please also update the corresponding spot in `pkg/datapath`, so that we keep track of the mark usage?